libs/btp/tool-header/src/components/tool-header/tool-header.component.ts

```diff
@@ -0,0 +1,257 @@
+/* eslint-disable @angular-eslint/no-input-rename */
+import { BooleanInput, coerceBooleanProperty } from '@angular/cdk/coercion';
+import { NgForOf, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ContentChild,
+    EventEmitter,
+    HostListener,
+    Input,
+    NgZone,
+    OnDestroy,
+    Output,
+    ViewEncapsulation,
+    computed,
+    inject
+} from '@angular/core';
+import { SearchFieldComponent } from '@fundamental-ngx/btp/search-field';
+import { FD_PRODUCT_SWITCH_COMPONENT } from '@fundamental-ngx/core';
+import { ButtonComponent } from '@fundamental-ngx/core/button';
+import { contentDensityObserverProviders } from '@fundamental-ngx/core/content-density';
+import { IconComponent } from '@fundamental-ngx/core/icon';
+import {
+    GlyphMenuAddonDirective,
+    MenuAddonDirective,
+    MenuComponent,
+    MenuInteractiveComponent,
+    MenuItemComponent,
+    MenuTitleDirective,
+    MenuTriggerDirective
+} from '@fundamental-ngx/core/menu';
+import {
+    OverflowExpandDirective,
+    OverflowItemRefDirective,
+    OverflowLayoutComponent,
+    OverflowLayoutItemDirective
+} from '@fundamental-ngx/core/overflow-layout';
+import { PopoverBodyDirective, PopoverComponent, PopoverControlComponent } from '@fundamental-ngx/core/popover';
+import { FdTranslatePipe } from '@fundamental-ngx/i18n';
+import { Subscription, first } from 'rxjs';
+import { ToolHeaderActionsDirective } from '../../directives/tool-header-actions.directive';
+import { ToolHeaderButtonDirective } from '../../directives/tool-header-button.directive';
+import { ToolHeaderElementDirective } from '../../directives/tool-header-element.directive';
+import { ToolHeaderGroupDirective } from '../../directives/tool-header-group.directive';
+import { ToolHeaderLogoDirective } from '../../directives/tool-header-logo.directive';
+import { ToolHeaderUserDirective } from '../../directives/tool-header-user.directive';
+import { ToolHeaderComponentClass } from '../../tool-header-component.class';
+
+export type FdbToolHeaderMode = 'desktop' | 'tablet' | 'mobile';
```

Comment:
do you need `desktop` since it's the default one?
The Tool Layout was merged with this type:
```
export type ToolLayoutMode = '' | 'tablet' | 'phone';
```
I think you wanted to unify them?

Reply:
I will make my type compatible with yours and will unify approach later. Can't fit in timeframes now